### PR TITLE
Bug 1447537: Adjust document title when viewing particular intermittent

### PR DESCRIPTION
**Pull request by smuny**
@sarah-clements  @camd 
1. Added `react-helmet` package.
2. Added `Helmet` fragment in `BugDetailsView.jsx`.

### CODE REVIEW

File: ui/intermittent-failures/BugDetailsView.jsx [hunk 1]
```diff
@@ -136,6 +137,10 @@ const BugDetailsView = props => {
       header={
         <React.Fragment>
           <Row>
+          <Helmet>
+            <meta charSet="utf-8" />
```
Comment (sarah-clements) [inline ui/intermittent-failures/BugDetailsView.jsx]: Why did you add a meta tag here? I don't think this would need to be in the row.

File: package.json [hunk 2]
```diff
@@ -68,7 +69,8 @@
     "taskcluster-lib-urls": "12.0.0",
     "victory": "33.0.5",
     "webpack": "4.39.3",
-    "webpack-cli": "3.3.8"
+    "webpack-cli": "3.3.8",
+    "yarn": "1.19.1"
```
Comment (sarah-clements) [inline package.json]: The only thing that should have changed is the addition of react-helmut, so you'll need to fix this file and the yarn.lock file. To resolve this you'll need to rebase onto the latest master - don't pull and merge. When you push your branch here do a force push with the `--force` flag.

Comment (smuny) [reply package.json]: I can do that. 

I noticed that this project on master used yarn to install and start so on my fork and branch, it was the latest yarn so I updated it because it gave me errors. I’m not sure what your thoughts were on this. I just read errors and proceeded to follow it.

File: ui/intermittent-failures/BugDetailsView.jsx (hunk 1 above)
Comment (smuny) [reply ui/intermittent-failures/BugDetailsView.jsx]: I’ll make these changes! Thanks for reviewing. I kept it there but I’ll remove it!

File: package.json (hunk 2 above)
Comment (sarah-clements) [reply package.json]: Was this what you saw?
Screen Shot 2019-10-09 at 10 03 58 AM
When you upgrade your local version of yarn, you should be sure to only update your system and not the code base files. Running `brew upgrade yarn` has the desired effect. If you used `yarn` to update itself, that is likely why it also updated `package.json` and `yarn.lock` files. 

We use a bot to manage dependency upgrades so that's why we don't want any upgrades done as part of an unrelated pull request.

Comment (sarah-clements) [reply package.json]: I see merge conflicts now, so you'll need to rebase on master and resolve the conflicts. Be sure to run `yarn` to catch any changes to the `package.json` file.

Comment (smuny) [reply package.json]: Sounds great! Thank you for clarifying. 

Screen Shot 2019-10-09 at 2 55 13 PM

Comment (smuny) [reply package.json]: That was the error message I got. It said to `yarn add react-helmet`

Comment (sarah-clements) [reply package.json]: Right, so you _do_ want to add `yarn add react-helmut`, you _do not_ want to update yarn with `yarn add yarn` - only via something like homebrew. I'm also seeing a lot of other packages updated in the lockfile.... did you also run `yarn upgrade`?

Comment (sarah-clements) [reply package.json]: I think the easiest way to clean this up is to do a soft reset of your commit and remove/discard all files you had commited except for the bugdetails view, see https://stackoverflow.com/questions/12481639/remove-files-from-git-commit

Then do `yarn add react-helmut` and commit the changes.

Don't do a force push to his branch unless you know that was successful, otherwise it'll override your work. You can make a copy of this branch first as a safeguard.

Comment (smuny) [reply package.json]: Thank you for guiding me. Yeah I did do `yarn add yarn` but I should've read more into what that actually does instead. I'll make a new branch and do a soft reset.

**Comment by smuny** [comment]
@sarah-clements 
1. Changed yarn update and added react-helmet on `react-helmet` branch

**Comment by sarah-clements** [comment]
> @sarah-clements
> 
>     1. Changed yarn update and added react-helmet on `react-helmet` branch

You still have merge conflicts you need to resolve - with the package.json file :)

**Comment by sarah-clements** [comment]
Almost there :) You need to get all tests to pass before this can be merged. If you look at the Project Contribution page on the Outreachy website (under "A few tips") there are links to several pages in our docs. These are commands you need to run in your terminal then commit to your branch.

**Comment by smuny** [comment]
Hi, sorry to bother you again. There were 12 errors that was unrelated to
my bug. How do I fix this? Also, I did the step with prettier and
formatting. Sorry sorry, it’s probably a simple fix that I’m unfamiliar
with.

**Comment by sarah-clements** [comment]
> Hi, sorry to bother you again. There were 12 errors that was unrelated to my bug. How do I fix this? Also, I did the step with prettier and formatting. Sorry sorry, it’s probably a simple fix that I’m unfamiliar with.

No need to apologize - everyone is new to something at some point :) I'm only seeing the prettier errors - I clicked on the details link above for "continuous-integration/travis-ci/pr — The Travis CI build failed". Can you paste a screen shot of what errors you're seeing?

**Comment by smuny** [comment]
[image: Screen Shot 2019-10-14 at 2.33.50 PM.png]

Thank you for understanding

**Comment by sarah-clements** [comment]
> [image: Screen Shot 2019-10-14 at 2.33.50 PM.png] Thank you for understanding

I'm not actually seeing a screen shot. Can you try again?

**Comment by smuny** [comment]
I sent it as an attachment this time. Let me know if it works!

**Comment by sarah-clements** [comment]
> I sent it as an attachment this time. Let me know if it works!

I'm not sure what you mean. You can drag or attach and image here in github and it will show up in the comments section. If you click 'preview' before commenting, it'll show you whether or not it was successful. (I still don't see it - I suggest searching github docs if you're having problems).

**Comment by smuny** [comment]
Screen Shot 2019-10-14 at 2 33 50 PM

**Comment by sarah-clements** [comment]
hrmm, I don't see those on latest master. Try rebasing again?

File: ui/intermittent-failures/BugDetailsView.jsx [hunk 3]
```diff
@@ -141,6 +142,10 @@ const BugDetailsView = props => {
       header={
         <React.Fragment>
           <Row>
+            <Helmet>
+              <meta charSet="utf-8" />
+              <title>{summary}</title>
```
Comment (sarah-clements) [inline ui/intermittent-failures/BugDetailsView.jsx]: I've tested this and it's certainly an improvement but it only contains partial information; the bug number, so `bug 12345 - whatever the summary is` should also be included. 

Also, I don't think you answered my question about the meta tag and why that's being used here. :)

Comment (smuny) [reply ui/intermittent-failures/BugDetailsView.jsx]: The meta tag is created uniquely for that page, in this case, for the particular bug page.Interestingly, I googled that the meta tag won't be duplicated but rather it'll override so for an example, if a user was to be on the particular bug page, the meta tag will be unique to that page rather than the main index page.
A question I have is, do I create a react-helmet on the index and then create parent and children relationship with meta tags?

Comment (smuny) [reply ui/intermittent-failures/BugDetailsView.jsx]: Also, I tried to replace the tab to show `bug 12345 - whatever the summary is` with 
            `<Helmet>
              <title>Bug {bug} - {summary}</title>
            </Helmet>`
and it wasn't showing up for me. Am I missing something?

Comment (sarah-clements) [reply ui/intermittent-failures/BugDetailsView.jsx]: > Also, I tried to replace the tab to show `bug 12345 - whatever the summary is` with
> `<Helmet> <title>Bug {bug} - {summary}</title> </Helmet>`
> and it wasn't showing up for me. Am I missing something?

You need to use a template literal.

Comment (sarah-clements) [reply ui/intermittent-failures/BugDetailsView.jsx]: > The meta tag is created uniquely for that page, in this case, for the particular bug page.Interestingly, I googled that the meta tag won't be duplicated but rather it'll override so for an example, if a user was to be on the particular bug page, the meta tag will be unique to that page rather than the main index page.
> A question I have is, do I create a react-helmet on the index and then create parent and children relationship with meta tags?

I don't know what the benefit of that would be - the docs aren't very clear. We only want to change the title for bug details view so I think this is fine.

Comment (smuny) [reply ui/intermittent-failures/BugDetailsView.jsx]: Sorry, only because I’ve been trying different things out. I did and go through with using a back ticks, dollar sign and curly braces for expressions. On the tab, it shows as it is. How I initially wrote the expression in the curly braces worked. Previously having ‘{bug}’ and ‘{summary}’ worked but then when I refreshed the page, it went blank. I’m hoping to figure it out a bit more tonight but if not, I would like to know what you think. I asked the public chat if anyone knew react helmet but I didn’t get an answer so I’m thinking it fairly new everyone. 

Other than adding the bug detail number, I’m really thankful and appreciative to have you work so patiently.

Comment (sarah-clements) [reply ui/intermittent-failures/BugDetailsView.jsx]: If a page goes blank, then you'll need to open the devtools panel and look in the console for errors. This is an important tool for development so it's a good thing to get familiar with :) Also, frequently running `yarn lint --fix` in your terminal will catch errors that could be causing the page to break (go blank).

For the title, what you'd need to do is this: `{`Bug ${bug} - ${summary}`}`. Having to wrap a template literal/string in curly brackets is an aspect of using jsx in React.

**Comment by sarah-clements** [comment]
This isn't ready for review yet - there are still merge conflicts. If you look at the files tab, you should only see changes relevant to this bug. You need to be sure you've set the upstream remote so that you can sync your fork to upstream and rebase: https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork
